app/routes.py

```diff
@@ -1,2 +1,344 @@
+                            completed_at=request_body["completed_at"],
+                            is_complete=True
+                            )
+            db.session.add(new_task)
+            db.session.commit()
+
+
+            return jsonify({"task":
+                {
+                "id" : new_task.task_id,
+                "title" : new_task.title,
+                "description" : new_task.description,
+                "is_complete" : new_task.is_complete
+                    }
+                }), 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        
+        return_dict = {"task":
+            {
+            "id" : task.task_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+                }
+            }
+
+        if task.goal_id:
+            return_dict["task"]["goal_id"] = task.goal_id
+
+        return return_dict, 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+
+        if not task.completed_at:
+        
+            task.title = request_body["title"]
+            task.description = request_body["description"]
+
+            db.session.commit()
+
+        elif task.completed_at:
+            task.title = request_body["title"]
+            task.description = request_body["description"]
+            task.is_complete = True
+
+            db.session.commit()
+
+        return jsonify(
+                {
+                "task":
+                {
+                "id" : task.task_id,
+                "title" : task.title,
+                "description" : task.description,
+                "is_complete" : task.is_complete
+                    }
+                }
+                ), 200
+
+
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify(
+            {
+                'details': (f'Task {task.task_id} \"{task.title}\" successfully deleted')
+            }
+
+        ), 200
+
+
+PATH = "https://slack.com/api/chat.postMessage"
+API_KEY = os.environ.get("SLACK_KEY")
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def patch_it_up(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    if request.method == "PATCH":
+        task.completed_at = datetime.utcnow()
+        task.is_complete = True
+        db.session.commit()
+
+        #we put the auth here
+        header = {"authorization" : f"Bearer {API_KEY}"
+        }
+        post_body = {"channel" : "slack-api-test-channel",
+        "text": f"Someone just completed the task {task.title}"
+        }
+        
+        requests.post(PATH, headers=header, json=post_body)
```

Comment:
Consider putting this part of the code into _its own_  helper function. All the mark complete endpoint really needs to know how to do is call a method that can post to Slack with the message to post. The rest of the details could be hidden in the helper, and potentially called from multiple places!